pallets/collator-selection/src/tests.rs

```diff
@@ -247,7 +259,31 @@ fn session_management_works() {
 		initialize_to_block(20);
 		assert_eq!(SessionChangeBlock::get(), 20);
 		// changed are now reflected to session handlers.
-		assert_eq!(SessionHandlerCollators::get(), vec![1, 2, 3]);
+		assert_eq!(SessionHandlerCollators::get(), vec![1, 2, 4]);
+	});
+}
+
+#[test]
+fn boot_mechanism() {
+	new_test_ext().execute_with(|| {
+		// add a new collator
+		assert_ok!(CollatorSelection::register_as_candidate(Origin::signed(3)));
+		assert_ok!(CollatorSelection::register_as_candidate(Origin::signed(4)));
+		assert_eq!(CollatorSelection::candidates().len(), 2);
+		initialize_to_block(21);
+		assert_eq!(CollatorSelection::boot_block(), 20);
+		assert_eq!(SessionChangeBlock::get(), 20);
+		// 4 authored this block, gets to stay 3 was booted
```

Comment:
is "being booted" == "being kicked", terminology-wise?

Reply:
ya, tbh I was gonna bring up that I was having trouble naming this variable lol

Reply:
I would call kicking or yanking.